minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/util/EnvironmentVariableMap.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.minikube.util;
+
+import java.util.HashMap;
+import java.util.List;
+
+/** Represents a environment map from environment variable keys to values. */
+public class EnvironmentVariableMap extends HashMap<String, String> {
```

Comment:
This class itself might be a little overkill for what we're trying to do. 
- Perhaps a simple parser that reads a list of strings and outputs a map is all we need. MinikubeEnvParser or something. Then we get a standard Map instance to manipulate in other places.
- Not super on board with the idea of extending Map. Making this an input parameter to "setEnvironment" doesn't seem to allow us to use a simple Map in CommandExecutor if we wanted to. The extended class itself doesn't offer too much in terms of functionality and would probably (subjective of course) more readable if it just kept an instance of a HashMap.
- I would probably avoid the custom runtime exceptions, if they're not going to be checked then we might as well use one of the well known existing Runtime exceptions like `IllegalArgumentException`.

Reply:
Okay changed the class to MinikubeDockerEnvParser to just be a static parser.